sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1456,6 +1545,59 @@ all the nodes have finished execution.
 
 The queue should be associated with a device and context that are the same
 as the device and context used on creation of the graph.
+
+| 
+[source,c++]
+----
+event queue::ext_oneapi_external_event(const event& depEvent);
+----
+
+|Create and return a graph external event representing the execution of
+`depEvent`. This function should only be called on a queue which is in the
+recording state.
+
+For more information on external events see <<event-class-modifications, the
+"Event Class Modifications" section>>.
+
+Parameters:
+
+* `depEvent` - The event which this external event depends on. The
+external event will be considered complete when all of its dependencies are
+complete.
+
+Returns: The external event representing this point of execution in the graph.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if the queue is not 
+  in recording mode. 
+
+| 
+[source,c++]
+----
+event queue::ext_oneapi_external_event(const std::vector<event>& depEvents);
+----
+
+|Create and return a graph external event representing the execution of
+`depEvents`. This function should only be called on a queue which is in the
+recording state.
+
+For more information on external events see <<event-class-modifications, the
+"Event Class Modifications" section>>.
+
+Parameters:
+
+* `depEvents` - List of events which this external event depends on. The
+external event will be considered complete when all of its dependencies are
+complete.
+
+Returns: The external event representing this point of execution in the graph.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if the queue is not 
+  in recording mode. 
```

Comment:
I think these two member functions are intended to be used when recording a graph.  How realistic is it that the caller will know the `depEvents`?  I thought the usage model is that the caller starts recording a queue, and then calls some non-graph-aware function, which puts commands on the queue.  Since the called function is not aware it is being recorded, it can't call this new graph-specific API.  However, the `depEvents` are probably internal details of the function, so the caller isn't aware of them.

I wonder if it would be better to have a new API that returns an external event which represent a dependency on all the commands that have been recorded to the queue so far.  This would allow a caller to record a dependency point inside a recorded graph without knowing the specific `depEvents`.

I also think the `ext_oneapi_external_event` does not work well when recording an in-order queue.  In this case, there typically is not any `depEvents`.  Will the external event returned by this function have an implicit dependency on all previous commands in this case?